### PR TITLE
executor, expression: fix floor and ceil when handle int overflow (#3980)

### LINKED ISSUE #3980: floor and ceil is not consistent with MySQL
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
``` sql
select floor(18446744073709551615), ceil(18446744073709551615);
select floor(18446744073709551615.1233), ceil(18446744073709551615.1233);
```

2. What did you expect to see?
``` sql
mysql> select floor(18446744073709551615), ceil(18446744073709551615);
+-----------------------------+----------------------------+
| floor(18446744073709551615) | ceil(18446744073709551615) |
+-----------------------------+----------------------------+
|        18446744073709551615 |       18446744073709551615 |
+-----------------------------+----------------------------+
1 row in set (0.00 sec)

mysql> select floor(18446744073709551615.1233), ceil(18446744073709551615.1233);
+----------------------------------+---------------------------------+
| floor(18446744073709551615.1233) | ceil(18446744073709551615.1233) |
+----------------------------------+---------------------------------+
|             18446744073709551615 |            18446744073709551616 |
+----------------------------------+---------------------------------+
1 row in set (0.00 sec)
```

3. What did you see instead?
``` sql
mysql> select floor(18446744073709551615), ceil(18446744073709551615);
+-----------------------------+----------------------------+
| floor(18446744073709551615) | ceil(18446744073709551615) |
+-----------------------------+----------------------------+
|        -9223372036854775808 |       -9223372036854775808 |
+-----------------------------+----------------------------+
1 row in set (0.00 sec)

mysql> select floor(18446744073709551615.1233), ceil(18446744073709551615.1233);
+----------------------------------+---------------------------------+
| floor(18446744073709551615.1233) | ceil(18446744073709551615.1233) |
+----------------------------------+---------------------------------+
|             -9223372036854775808 |            -9223372036854775808 |
+----------------------------------+---------------------------------+
1 row in set (0.00 sec)
```

4. What version of TiDB are you using (`tidb-server -V`)?
Git Commit Hash: d0dcb5b0a6a518a58ea64867051ca2b7f3861f8f
UTC Build Time: 2017-08-01 03:45:12

### PR DESCRIPTION
**Pull request by XuHuaiyu**
fix #3980

### CODE REVIEW

File: expression/builtin_math.go [hunk 1]
```diff
@@ -64,11 +65,11 @@ var (
 var (
 	_ builtinFunc = &builtinAbsSig{}
 
-	_ builtinFunc = &builtinCeilRealSig{}
-	_ builtinFunc = &builtinCeilIntSig{}
-
-	_ builtinFunc = &builtinFloorRealSig{}
-	_ builtinFunc = &builtinFloorIntSig{}
+	//_ builtinFunc = &builtinCeilRealSig{}
```
Comment (shenli) [inline expression/builtin_math.go]: Why comment these lines?

Comment (XuHuaiyu) [inline expression/builtin_math.go]: Because these sigs have been rewritten.

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason  @AndreMouche

File: executor/executor_test.go [hunk 2]
```diff
@@ -1611,6 +1611,10 @@ func (s *testSuite) TestMathBuiltin(c *C) {
 	result.Check(testkit.Rows("-10 -10"))
 	result = tk.MustQuery("select ceil(t.c_decimal), ceiling(t.c_decimal) from (select cast('-10.01' as decimal(10,1)) as c_decimal) as t")
 	result.Check(testkit.Rows("-10 -10"))
+	result = tk.MustQuery("select floor(18446744073709551615), ceil(18446744073709551615)")
```
Comment (AndreMouche) [inline executor/executor_test.go]: Should we add some annotation here or split `floor` from this test block since L1595 mentations ` for ceil/ceiling`?

**Comment by XuHuaiyu** [comment]
PTAL @winkyao

File: executor/executor_test.go [hunk 3]
```diff
@@ -1611,6 +1611,10 @@ func (s *testSuite) TestMathBuiltin(c *C) {
...
+	result.Check(testkit.Rows("18446744073709551615 18446744073709551615"))
+	result = tk.MustQuery("select floor(18446744073709551615.1233), ceil(18446744073709551615.1233)")
+	result.Check(testkit.Rows("18446744073709551615 18446744073709551616"))
```
Comment (AndreMouche) [inline executor/executor_test.go]: Could we add some test like `floor(-max_int),ceil(-max_int)`?

**Comment by XuHuaiyu** [comment]
@AndreMouche @winkyao

**Pull request merged**